lib/puppet/node.rb

```diff
@@ -45,10 +45,20 @@ def to_data_hash
       'environment' => environment.name.to_s,
     }
     result['classes'] = classes unless classes.empty?
-    result['parameters'] = parameters unless parameters.empty?
+    result['parameters'] = serializable_parameters unless serializable_parameters.empty?
```

Comment:
Here you call serialize_parameters twice if the list isn't empty, which will dup parameters twice throwing one copy away. How costly could that be. We should assign this to a local variable and then check if its  empty?

Reply:
Ah, yeah. Something like makes sense to do here:

```Ruby
serialized_params = serializable_parameters
result['parameters'] = serialized_params unless serialized_params.empty?
```